src/reducers/console.js

```diff
@@ -2,7 +2,11 @@ import {OrderedMap} from 'immutable';
 
 import {ConsoleEntry, ConsoleError} from '../records';
 
-const initialState = new OrderedMap();
+const initialState = {
+  history: new OrderedMap(),
+  previousHistoryIndex: 0,
+  previousConsoleInput: '',
+};
```

Comment:
Also, thinking about the high-level structure here—not a huge change from what you’ve got but I would probably model this as:

* `entries` (what you’re calling `history`, although either is fine)
* `currentInputEntryIndex` (or something—I don’t love this): I would make this `null` unless/until the user starts navigating up through previous entries.
* `nextConsoleEntry`: This would be what the user typed in the console before navigating up through previous entries; null if they have not navigated up.
* `currentInputValue`: Whatever is currently in the input editor. See my comment on the `<ConsoleInput>` component for motivation.

Another approach would be to store the `nextConsoleEntry` simply as the last item in the list of entries, presumably with an additional boolean property in `ConsoleEntry` to indicate that an entry is in-progress. Ultimately though I think storing it as its own entity is less complicated and no worse at modeling the essential state of the application.